executor/include/execution/executor/base_executor.hpp

```diff
@@ -4,22 +4,33 @@
 
 #pragma once
 
+#include <array>
+#include <execution/property.hpp>
+#include <execution/type_trait.hpp>
 #include <functional>
 #include <iostream>
+#include <memory>
 #include <stdexcept>
 #include <string>
-#include <array>
-#include <memory>
-
-#include <execution/property.hpp>
-#include <execution/type_trait.hpp>
-
 
-template <template <typename, typename, typename, typename> typename Derived,
-    typename Interface, typename Cardinality, typename Blocking,
-    typename ProtoAllocator>
+template <template <typename...> typename Derived, typename Interface,
+          typename Cardinality, typename Blocking, typename ProtoAllocator>
 class executor {
  public:
+  template <typename Executor,
+            typename execution::instance_of_base<Derived, Executor> = 0>
+  bool operator==(const Executor &) const noexcept {
+    return std::is_same_v<
+        Derived<Interface, Cardinality, Blocking, ProtoAllocator>, Executor>;
+  }
+
+  template <typename Executor,
+            typename execution::instance_of_base<Derived, Executor> = 0>
+  bool operator!=(const Executor &) const noexcept {
+    return std::is_same_v<
```

Comment:
o.O ?

Reply:
you should probably just negate your equivalence operator.

Reply:
Oops, Fixed 293f980bea87c88d2c7f862f8916d40dd8a32a69